DS/DSassignment1/ArrayList1.java

```diff
@@ -0,0 +1,203 @@
+import java.util.Arrays;
+
+public class ArrayList1 {
+	
+	public static int defaultSize = 10;
+	public static int size=0;
+	public static Object data[] = new Object[defaultSize];
+	
+	public static int index;
+	
+	
+	public  void add(Object value){
+		if(size == data.length ){
+			increaseCapacity();
+		}
+		data[size++] = value;
+	
+		
+	}
+	
+	public void merge(){
+		if((ArrayList2.size + size) >= data.length ){
+			increaseCapacity();
+		}
+		for(int i = 0; i < ArrayList2.size;i++)
+		add(ArrayList2.newData[i]) ;	
+		System.out.println("both list merged");
+		
+	}
+	public static void increaseCapacity(){
+		int newsize = data.length * 2;
+		data = Arrays.copyOf(data, newsize);
+	}
+	
+		public Object get(int i) {
+		if(index==-1){
+			return 0;
+		}
+		else
+		return data[i];
+		
+	}
+	
+	public  static void reverse() {
+		  if(data == null || data.length <= 1){
```

Comment:
Looking the complete code i don't think there will be a case when 'data' would be null.